### PR TITLE
Make athena views use only athena

### PR DESCRIPTION
**Pull request by rleighton**
## Before

Creating the Athena views used the table resolvers to create the views, coupling the views to our parsers.

## After

Since views are, by definition over existing tables, we can simply query Athena for the tables and columns and create the views that way.

The implementation has a generic part and a database specific part (that lists the tables and columns). This way adding new databases is easy by just implementing a "lister" interface.

I have moved this code under `log_analysis/datalake/views` . 

image

## Testing

- mage test:ci
- mage deploy + check

### CODE REVIEW

File: internal/log_analysis/datalake/views/views.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
+package views
```
Comment (rleighton) [inline internal/log_analysis/datalake/views/views.go:1]: This the core of the old code put into an abstract class so we can drive it from multiple databases.

**Review by alxarch — COMMENTED** [review]
This will definitely clean up our code base. Kudos!

**Review by alxarch — APPROVED** [review]

**Comment by alxarch** [comment]
Note that AWS SDK v2 _only provides `context.Context` methods_. We plan to transition gradually and having proper support for it throughout Panther will improve our code reliability and composability a lot.

File: internal/log_analysis/datalake/athena/athenaviews/athenaviews.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	"github.com/panther-labs/panther/internal/log_analysis/pantherdb"
+	"github.com/panther-labs/panther/pkg/awsathena"
+)
+
+var (
+	catalogName = "AwsDataCatalog"
+)
+
+type ViewMaker struct {
+	ctx          context.Context
```
[the remark this answers is not in the document]
Comment (rleighton) [reply internal/log_analysis/datalake/athena/athenaviews/athenaviews.go]: done

File: internal/log_analysis/datalake/athena/athenaviews/athenaviews.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		cols[i] = &athenaColumn{Column: *col, isPartition: false}
+	}
+	for i, col := range at.tableData.PartitionKeys {
+		cols[i+len(at.tableData.Columns)] = &athenaColumn{Column: *col, isPartition: true}
+	}
+	return cols
+}
+
+// CreateOrReplaceLogViews will update Athena with all views for the tables provided
+func (m *ViewMaker) CreateOrReplaceLogViews() error {
```
[the remark this answers is not in the document]
Comment (rleighton) [reply internal/log_analysis/datalake/athena/athenaviews/athenaviews.go]: done

File: internal/log_analysis/datalake/views/views.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+type Table interface {
+	DatabaseName() string
+	Name() string
+	Columns() []Column
+}
+
+type TableLister interface {
+	ListTables(databaseName string) (tables []Table, err error)
```
[the remark this answers is not in the document]
Comment (rleighton) [reply internal/log_analysis/datalake/views/views.go]: done

**Pull request merged**